langchain/src/chat_models/configurable.ts

```diff
@@ -0,0 +1,877 @@
+import {
+  BaseLanguageModelInput,
+  ToolDefinition,
+} from "@langchain/core/language_models/base";
+import {
+  BaseChatModel,
+  BaseChatModelParams,
+  type BaseChatModelCallOptions,
+} from "@langchain/core/language_models/chat_models";
+import { BaseMessage, type AIMessageChunk } from "@langchain/core/messages";
+import {
+  Runnable,
+  type RunnableBatchOptions,
+  RunnableBinding,
+  type RunnableConfig,
+  type RunnableToolLike,
+  ensureConfig,
+} from "@langchain/core/runnables";
+import {
+  AsyncGeneratorWithSetup,
+  IterableReadableStream,
+} from "@langchain/core/utils/stream";
+import {
+  type LogStreamCallbackHandlerInput,
+  type RunLogPatch,
+  type StreamEvent,
+} from "@langchain/core/tracers/log_stream";
+import { type StructuredToolInterface } from "@langchain/core/tools";
+import { CallbackManagerForLLMRun } from "@langchain/core/callbacks/manager";
+import { ChatResult } from "@langchain/core/outputs";
+
+// TODO: remove once `EventStreamCallbackHandlerInput` is exposed in core.
```

Comment:
Should we just expose it?

Reply:
You can also extract the types from the base class I think

Reply:
Planning on exposing but needs to be a separate PR